observability/WFLY-14854_add_opentelemetry_support.adoc

```diff
@@ -0,0 +1,76 @@
+ = Add OpenTelemetry Tracing Support to WildFly
+:author:            Jason Lee
+:email:             jasondlee@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+https://opentelemetry.io[OpenTelemetry] is an ambitious specification that aims to encapsulate APIs for tracing, metrics, and logging. With OpenTracing (which WildFly already supports) having shutdown in favor of OpenTelemetry, we need to migrate our support as well. This project will only cover the tracing aspects of OpenTelemetry. Any future support for the metrics and logging parts of the OpenTelemetry spec will need to be handled by a separate effort.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14854
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1688
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+* [X] Engineering
+* [X] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+* [ ] Managed domain
+* [x] OpenShift s2i
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* The server/domain administrator should be able to configure OpenTelemetry export and span processor options
+* Jakarta REST endpoints should be automatically traced
+* Adherence to the Observability Standard for Runtimes
+* Context propagation should work for both incoming and outgoing request (e.g., a `ClientRequestFilter`)
+
+=== Nice-to-Have Requirements
+
+* Feature parity with current OpenTracing support, if/where it makes sense.
+* Tracing of JDBC calls
+* Tracing of EJB calls
+
+=== Non-Requirements
+
+* Multiple span processor support (if desired, users should submit an RFE)
+* Metrics API support
+* Logging API support
+* Use of the OpenTracing shim to allow existing OpenTracing apps to continue to function without modification (OpenTracing will not be removed until and if MicroProfile dictates that we do)
```

Comment:
If having both things in the config won't work well, what will be done to prevent such a config?

Reply:
Having both (OpenTracing and OpenTelemetry) configured at the same time won't *break* anything, but the user will see traces exported from both systems. This removes OpenTracing, and should, perhaps, be added to the community docs?

```
jboss-cli.sh -c "/subsystem=microprofile-opentracing-smallrye:remove()"
jboss-cli.sh -c "/extension=org.wildfly.extension.microprofile.opentracing-smallrye:remove()"
```

Reply:
This has been added to the community docs.